### PR TITLE
Add a retry strategy for the subgraphs tests

### LINKED ISSUE #1128: Subgraph tests
### Subject of the issue
We have some tests in order to test the subgraphs' queries that we are using. However, these tests depend on a) network and b) an external resource, and therefore, they may occasionally fail. For example, test `tests/test_skills/test_apy_estimation_abci/test_apis.py::TestSubgraphs::test_top_n_pairs[UniswapSubgraph-uni_specs_pairs_extended]` fails in [main_workflow #4429](https://github.com/valory-xyz/open-autonomy/runs/7563825270?check_suite_focus=true).

There are 3 potential solutions:
1. The subgraphs cannot change once deployed, therefore we may skip the tests. That would be a problem if we updated a query incorrectly.
2. We may add a retry strategy to these tests.
3. We may mark all the tests that depend on network connectivity/external resources.

Solution (2.) might be the most appropriate at this point.

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Adds a retry strategy for the subgraphs tests which depend on the network and may fail occasionally.

## Fixes

Closes #1128.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: tests/test_skills/test_apy_estimation_abci/test_apis.py [hunk 1]
```diff
@@ -49,7 +50,7 @@
 SubgraphResponseType = Dict[str, ResponseItemType]
 
 
-def make_request(
+def __make_request_no_retries(
```
Comment (Karrenbelt) [inline tests/test_skills/test_apy_estimation_abci/test_apis.py:53]: for a function this does not matter, but please refrain from using double underscores generally, single underscore marks private and should be clear to other devs. Particularly can be annoying when something if used in methods without actual need for such, leads to name mangling during inheritance (you're basically saying, even child classes should not access this)

**Review by Karrenbelt — APPROVED** [review]
This should really be generalized and put into a class with generic helper functions, as I've seen this same logic implemented several places (and yes, I am also a "contributor" to that).

However we can do better than what we have so far done and use built-in libraries entirely. Please open a ticket if you agree with my suggestion, link this PR so we can replace and discard this implementation and others (like my own) after

```
import requests
from requests.adapters import HTTPAdapter, Retry

def requests_retry_session(...):
    s = requests.Session()
    retries = Retry(total=5, backoff_factor=0.1, status_forcelist=[404, ... ])
    s.mount('http://', HTTPAdapter(max_retries=retries))
    s.get(api_specs["url"])
```

https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#module-urllib3.util.retry
https://www.peterbe.com/plog/best-practice-with-retries-with-requests

**Comment by Adamantios** [comment]
> This should really be generalized and put into a class with generic helper functions, as I've seen this same logic implemented several places (and yes, I am also a "contributor" to that).
> 
> However we can do better than what we have so far done and use built-in libraries entirely. Please open a ticket if you agree with my suggestion, link this PR so we can replace and discard this implementation and others (like my own) after
> 
> ```
> import requests
> from requests.adapters import HTTPAdapter, Retry
> 
> def requests_retry_session(...):
>     s = requests.Session()
>     retries = Retry(total=5, backoff_factor=0.1, status_forcelist=[404, ... ])
>     s.mount('http://', HTTPAdapter(max_retries=retries))
>     s.get(api_specs["url"])
> ```
> 
> https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#module-urllib3.util.retry https://www.peterbe.com/plog/best-practice-with-retries-with-requests

I think we cannot because if you notice the implementation, we get status 200 and a specific error message in the returned dictionary which we need to handle.

**Review by DavidMinarsch — APPROVED** [review]

**Comment by Karrenbelt** [comment]
> > This should really be generalized and put into a class with generic helper functions, as I've seen this same logic implemented several places (and yes, I am also a "contributor" to that).
> > However we can do better than what we have so far done and use built-in libraries entirely. Please open a ticket if you agree with my suggestion, link this PR so we can replace and discard this implementation and others (like my own) after
> > ```
> > import requests
> > from requests.adapters import HTTPAdapter, Retry
> > 
> > def requests_retry_session(...):
> >     s = requests.Session()
> >     retries = Retry(total=5, backoff_factor=0.1, status_forcelist=[404, ... ])
> >     s.mount('http://', HTTPAdapter(max_retries=retries))
> >     s.get(api_specs["url"])
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#module-urllib3.util.retry https://www.peterbe.com/plog/best-practice-with-retries-with-requests
> 
> I think we cannot because if you notice the implementation, we get status 200 and a specific error message in the returned dictionary which we need to handle.

yes so we generalize it one step further and wrap this basic requests retry with back off method such that it is itself retried when specific message is returned.

**Pull request merged**